src/v/cloud_storage_clients/s3_client.h

```diff
@@ -243,4 +243,7 @@ class s3_client : public client {
     ss::shared_ptr<client_probe> _probe;
 };
 
+std::variant<client::delete_objects_result, rest_error_response>
+iobuf_to_delete_objects_result(iobuf&& buf);
```

Comment:
i know you didn't add this function here, but you are making it "public" so i believe the following comment is appropriate 

This function can have a better name.

* `x_to_*` usually means a type conversion/format of some sort, not the case here
* this method expects the input to be xml, the method doesn't say anything about the input expectations

something along the line of `parse_delete_objects_result` is much better name
* it tells what it is doing: parsing
* it tells the input expectations: delete objects result (from the context (ie the namespace location) we know that it must be a cloud storage delete objects result
* * **i'll correct myself** `parse_**s3**_delete_objects_result` is a better name, it won't work for azure but since it is in the same namespace it might confuse users
* iobuf is a detail and is part of the type declaration, we don't need it in the name. moreover, we can have multiple overloads for different type of "input carriers" and won't need to have dozen of function names